dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/SubsystemFeatureDefinitionImpl.java

```diff
@@ -467,21 +467,38 @@ public String toString() {
     /*
      * (non-Javadoc)
      *
-     * @see com.ibm.ws.kernel.feature.provisioning.ProvisioningFeatureDefinition#getIconFiles()
+     * @see com.ibm.ws.kernel.feature.provisioning.ProvisioningFeatureDefinition#getIcons()
      */
     @Override
     public Collection<String> getIcons() {
-        Collection<String> icons = new ArrayList<String>();
+        Collection<String> result = new ArrayList<String>();
         String iconHeader = getHeader("Subsystem-Icon");
         if (iconHeader != null) {
-            String[] iconHeaders = iconHeader.split(",");
-            for (String iconDirective : iconHeaders) {
-                String[] iconDirectives = iconDirective.split(";");
-                // Assume the first part is always the file URL
-                icons.add(iconDirectives[0].trim());
+            String[] icons = iconHeader.split(",");
+            for (String icon : icons) {
+                String[] iconAttrs = icon.split(";");
+                // icon has form "<iconFile>[; size=n]"
+                result.add(iconAttrs[0].trim());
+            }
+        }
+        String epIconsHeader = getHeader("Subsystem-Endpoint-Icons");
+        if (epIconsHeader != null) {
+            String epName = "";
+            String[] epIcons = epIconsHeader.split(",");
+            for (String epIcon : epIcons) {
+                String[] epIconAttrs = epIcon.split(";");
+                if (epIconAttrs[0].indexOf("=") >= 0) {
+                    // epIcon has form "<endpointName>=<iconFile>
+                    String[] epNameAndIconFile = epIcon.split("=");
+                    epName = epNameAndIconFile[0].trim();
+                    result.add(epName + "/" + epNameAndIconFile[1].trim());
+                } else {
+                    // epIcon has form "<iconFile>[; size=n]"
+                    result.add(epName + "/" + epIconAttrs[0].trim());
```

Comment:
What if epName is never set (remains "")?  This will add a path with a proceeding "/".  Will that proceeding "/" cause issues?  Perhaps add the path separator  if the `if (epIconAttrs[0].indexOf("=") >= 0) {` branch:
```
    epName =  epNameAndIconFile[0].trim() + "/";
    result.add(epName + epNameAndIconFile[1].trim());
} else {
    // epIcon has form "<iconFile>[; size=n]"
    result.add(epName + epIconAttrs[0].trim());
```

Reply:
Changes completed.